pages/_document.js

```diff
@@ -147,6 +147,37 @@ const resetStyles = `
   }
 `
 
+const offlineScript = `
+if (
+  typeof window !== 'undefined' &&
+  'serviceWorker' in navigator
+) {
+  navigator.serviceWorker.register('/sw.js', {
+    scope: './'
+  }).then(function(reg) {
+    reg.onupdatefound = function() {
+      const installingWorker = reg.installing
+
+      installingWorker.onstatechange = function() {
+        switch (installingWorker.state) {
+          case 'installed':
+            if (navigator.serviceWorker.controller) {
+              console.log('New or updated content is available.')
+            } else {
+              console.log('Content is now available offline!')
```

Comment:
Sidenote: This would be great to show in the UI somewhere, especially in the docs, that'd be an awesome followup PR!

Reply:
After merging this, let's make an issue to discuss it :)